c7n/resources/rdscluster.py

```diff
@@ -732,24 +732,32 @@ class PendingMaintenance(Filter):
                 resource: rds-cluster
                 filters:
                   - pending-maintenance
+                  - type: value
+                    key: '"c7n:PendingMaintenance".PendingMaintenanceActionDetails[].Action'
+                    op: intersect
+                    value:
+                      - system-update
     """
 
-    schema = type_schema('pending-maintenance')
+    annotation_key = 'c7n:PendingMaintenance'
+    schema = type_schema('pending-maintenance', rinherit=ValueFilter.schema)
     permissions = ('rds:DescribePendingMaintenanceActions',)
 
     def process(self, resources, event=None):
         client = local_session(self.manager.session_factory).client('rds')
 
         results = []
-        pending_maintenance = set()
+        pending_maintenances = []
         paginator = client.get_paginator('describe_pending_maintenance_actions')
         for page in paginator.paginate():
-            pending_maintenance.update(
-                {action['ResourceIdentifier'] for action in page['PendingMaintenanceActions']}
-            )
+            for action in page['PendingMaintenanceActions']:
+                pending_maintenances.append(action)
 
         for r in resources:
-            if r['DBClusterArn'] in pending_maintenance:
-                results.append(r)
+            for action in pending_maintenances:
+                if r['DBClusterArn'] == action['ResourceIdentifier']:
+                    r[self.annotation_key] = action
+                    results.append(r)
+                    break
```

Comment:
@sgungi-intuit what if there are more than 1 pending maintenance actions for a resource? This code will only capture the 1st action. 

Additionally, maybe we can avoid the nested for loop (n^2 complexity) by doing something like this
```
    def process(self, resources, event=None):
        client = local_session(self.manager.session_factory).client('rds')

        results = []
        resource_maintenances = {}
        paginator = client.get_paginator('describe_pending_maintenance_actions')
        for page in paginator.paginate():
            for action in page['PendingMaintenanceActions']:
                resource_maintenances.setdefault(action['ResourceIdentifier'], []).append(action)

        for r in resources:
            pending_maintenances = resource_maintenances.get(r['DBInstanceArn'], [])
            if len(pending_maintenances) > 0:
                r[self.annotation_key] = pending_maintenances
                results.append(r)

        return results
```

Reply:
Updated code and ran test case.